src/overlays/actors/ovl_En_Hintnuts/z_en_hintnuts.c

```diff
@@ -27,61 +50,519 @@ const ActorInit En_Hintnuts_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hintnuts/func_80A582B8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hintnuts/func_80A58320.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hintnuts/func_80A584DC.s")
+static ColliderCylinderInit cylinderInitData = {
+    { COLTYPE_UNK6, 0x00, 0x09, 0x39, 0x10, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x00, 0x01, 0x01 },
+    { 18, 32, 0, { 0, 0, 0 } }
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hintnuts/func_80A58608.s")
+static CollisionCheckInfoInit collisionCheckInfoInit = {
+    0x01,
+    0x0012,
+    0x0020,
+    0xFE,
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hintnuts/EnHintnuts_Update.s")
+static s16 puzzleCounter = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hintnuts/func_80A58838.s")
+static InitChainEntry initChain[] = {
+    ICHAIN_F32(gravity, -1, ICHAIN_CONTINUE),
+    ICHAIN_S8(naviEnemyId, 10, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_4C, 2600, ICHAIN_STOP),
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Hintnuts/EnHintnuts_Draw.s")
+void EnHintnuts_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnHintnuts* this = THIS;
+    s32 pad;
+
+    Actor_ProcessInitChain(&this->actor, initChain);
+    if (this->actor.params == 0xA) {
+        this->actor.flags &= ~5;
+    } else {
+        ActorShape_Init(&this->actor.shape, 0x0, ActorShadow_DrawFunc_Circle, 35.0f);
+        SkelAnime_Init(globalCtx, &this->skelAnime, &D_060023B8, &D_06002F7C, this->limbDrawTable,
+                       this->transitionDrawTable, 10);
+        Collider_InitCylinder(globalCtx, &this->collider);
+        Collider_SetCylinder(globalCtx, &this->collider, &this->actor, &cylinderInitData);
+        func_80061ED4(&this->actor.colChkInfo, NULL, &collisionCheckInfoInit);
+        Actor_SetTextWithPrefix(globalCtx, &this->actor, (this->actor.params >> 8) & 0xFF);
+        this->textIdCopy = this->actor.textId;
+        this->actor.params &= 0xFF;
+        puzzleCounter = 0;
+        if (this->actor.textId == 0x109B) // "Please forgive me, master!..."
+        {
+            if (Flags_GetClear(globalCtx, 0x9) != 0) {
+                Actor_Kill(&this->actor);
+                return;
+            }
+        }
+        EnHintnuts_SetupWait(this);
+        Actor_SpawnAttached(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_HINTNUTS, this->actor.posRot.pos.x,
+                            this->actor.posRot.pos.y, this->actor.posRot.pos.z, 0, this->actor.posRot.rot.y, 0, 0xA);
+    }
+}
+
+void EnHintnuts_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnHintnuts* this = THIS;
+    if (this->actor.params != 0xA) {
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+    }
+}
+
+void EnHintnuts_HitByNutsball1(EnHintnuts* this, GlobalContext* globalCtx) {
```

Comment:
I thought NutsBall seemed like a good choice, since ACTOR_EN_NUTSBALL is the name of the actor. i'm happy to change it to ScrubProjectile though if we want to prioritize clarity

Reply:
yeah for sure, I know why you picked that name. just think the clarity is more important. though we can probably just rename it when we decide to drop the original actor names